clouddriver-core/src/main/groovy/com/netflix/spinnaker/clouddriver/cache/OnDemandAgent.java

```diff
@@ -65,6 +69,28 @@ public OnDemandResult(String sourceAgentType, CacheResult cacheResult, Map<Strin
     }
   }
 
+  /*
+   * WARNING: this is an interim solution while cloud providers write their own ways to derive monikers.
```

Comment:
Worth logging an actual warning?

Reply:
I think so, when we migrate each cloud provider off. Before then it will just generate noise.